### PR TITLE
Fastify sdk contrib

### PR DESCRIPTION
**Pull request by jorgevrgs**
*Issue #, if available:*

*Description of changes:*

Add X-Ray plugin for Fastify

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by willarmiros** [comment]
Hi @jorgevrgs,

Thanks for contributing this instrumentation! Do you have any vision or plan for longer-term support for the instrumentation? Generally speaking we do not maintain the instrumentations in the `sdk_contrib`, so it would be nice to know someone with more expertise on the framework is able to help out with customer issues if they come in.

Also, we are advising newcomers to X-Ray to try out using OpenTelemetry JS, which already has an [instrumentation for fastify](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-fastify). So we also don't want to be unnecessarily duplicating instrumentation options if the OpenTelemetry one would be suitable. Let me know what you think.

**Comment by jorgevrgs** [comment]
> Hi @jorgevrgs,
> 
> Thanks for contributing this instrumentation! Do you have any vision or plan for longer-term support for the instrumentation? Generally speaking we do not maintain the instrumentations in the `sdk_contrib`, so it would be nice to know someone with more expertise on the framework is able to help out with customer issues if they come in.
> 
> Also, we are advising newcomers to X-Ray to try out using OpenTelemetry JS, which already has an [instrumentation for fastify](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-fastify). So we also don't want to be unnecessarily duplicating instrumentation options if the OpenTelemetry one would be suitable. Let me know what you think.

I checked the contribution guide about the support requirement. Yes, I can help you to bring support, but if you have someone in mind please invite them. It would be great to have more people contributing here.

**Comment by willarmiros** [comment]
Hi @jorgevrgs can you fix the failing CI tests? Then I can take a look

File: sdk_contrib/fastify/NOTICE.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+AWS X-Ray SDK Hapi for JavaScript
```
Comment (willarmiros) [inline sdk_contrib/fastify/NOTICE.txt:1]: Fastify

File: sdk_contrib/fastify/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "singleQuote": true
```
Comment (willarmiros) [inline sdk_contrib/fastify/.prettierrc:2]: I think we don't use prettier, can you remove this file?

File: sdk_contrib/fastify/package.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+  "name": "aws-xray-sdk-fastify",
+  "version": "3.3.6",
```
Comment (willarmiros) [inline sdk_contrib/fastify/package.json:3]: 3.3.7 now :)

File: sdk_contrib/fastify/package.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    "xray",
+    "x-ray",
+    "x ray"
+  ],
+  "directories": {
+    "lib": "lib"
+  },
+  "license": "Apache-2.0",
+  "repository": "https://github.com/aws/aws-xray-sdk-node/tree/sdk_contrib/fastify",
+  "devDependencies": {
```
Comment (willarmiros) [inline sdk_contrib/fastify/package.json:42]: Shouldn't be any dev dependencies, they should be hoisted

File: sdk_contrib/fastify/package.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (willarmiros) [inline sdk_contrib/fastify/package.json:1]: There should be no `package-lock.json` in this directory.

File: sdk_contrib/fastify/test/unit/xray.test.js [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+      await app.register(xrayFastifyPlugin, {
+        segmentName: 'test segment',
+        capturePromises: true,
+      });
+      expect(captureSpy).to.be.calledOnce;
+    });
+  });
+
+  describe('#createRequestHandler', function () {
+    const h = {
```
Comment (willarmiros) [inline sdk_contrib/fastify/test/unit/xray.test.js:138]: Remove?

**Review by willarmiros — COMMENTED** [review]
Looks like there are still some errors, possibly related to the lerna config. I would recommend bootstrapping & running tests locally to be sure they pass

**Comment by jorgevrgs** [comment]
> Looks like there are still some errors, possibly related to the lerna config. I would recommend bootstrapping & running tests locally to be sure they pass

Checking it, thanks for the info. Fixing ASAP.

File: sdk_contrib/fastify/package.json (hunk 4 above)
Comment (jorgevrgs) [reply sdk_contrib/fastify/package.json]: Working on this change

Comment (jorgevrgs) [reply sdk_contrib/fastify/package.json]: I have tried `npx lerna run test --scope=aws-xray-sdk-fastify` and now it seems to pass all the tests.

**Comment by willarmiros** [comment]
Hmm seems to still be failing - try running the same commands that the CI runs (found here: https://github.com/aws/aws-xray-sdk-node/blob/f432b2d099298af9f223ef03c781967a227eb481/.github/workflows/pr-build.yml#L45-L47) locally *from the root of the repo*. I guess that's the best way to avoid these snags

**Comment by jorgevrgs** [comment]
> 

Getting this error:

`lerna ERR! npm run test exited 1 in 'aws-xray-sdk-mysql'
lerna WARN complete Waiting for 4 child processes to exit. CTRL-C to exit immediately.`

**Comment by jorgevrgs** [comment]
> Hmm seems to still be failing - try running the same commands that the CI runs (found here:
> 
> https://github.com/aws/aws-xray-sdk-node/blob/f432b2d099298af9f223ef03c781967a227eb481/.github/workflows/pr-build.yml#L45-L47
> 
> ) locally _from the root of the repo_. I guess that's the best way to avoid these snags

Done:


image

**Review by willarmiros — APPROVED** [review]
LGTM! Thanks for getting the CI to green and addressing other comments

**Pull request merged**